### PR TITLE
FlexHTx with Codegen 

### PR DESCRIPTION
**Pull request by danielhstahl**

### CODE REVIEW

File: devtools/src/codegen/flexhtx/mod.rs [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+    }
+}
+
+pub(crate) fn output(output: usize) -> Output {
+    Output {
+        gate: Gate {
+            enable: format!("DGain_{}_0_status", output + 1),
+            gain: Some(format!("DGain_{}_0", output + 1)),
+        },
+        /*routing: (0..7usize) //input number
```
Comment (danielhstahl) [inline devtools/src/codegen/flexhtx/mod.rs:51]: some attempts to modify the output to conform with HTx additions (eg, the matrix routing)

File: devtools/src/codegen/flexhtx/mod.rs [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+        invert_addr: format!("polarity_out_{}_0", 1 + output),
+        peq: (1..=10usize)
+            .rev()
+            .map(|index| format!("PEQ_{}_{}", output + 1, index))
+            .collect(),
+        xover: Some(Crossover {
+            peqs: [1, 5]
+                .iter()
+                .map(|group| format!("BPF_{}_{}", output + 1, group))
+                .chain(
```
Comment (danielhstahl) [inline devtools/src/codegen/flexhtx/mod.rs:72]: In addition to BM_BPF_1_1 , there is a BM_BPF_ALL_1 and BM_BPF_ALL_5.  Not sure how to handle those.

File: devtools/src/codegen/flexhtx/mod.rs [hunk 3]
```diff
@@ -0,0 +1,121 @@
...
+                        .iter()
+                        .map(|group| format!("BM_BPF_{}_{}", output + 1, group)),
+                )
+                .collect(),
+        }),
+        compressor: Some(Compressor {
+            bypass: format!("COMP_{}_0_status", output + 1),
+            threshold: format!("COMP_{}_0_threshold", output + 1),
+            ratio: format!("COMP_{}_0_ratio", output + 1),
+            //knee: Some(format!("COMP_{}_0_knee", output + 1)),
```
Comment (danielhstahl) [inline devtools/src/codegen/flexhtx/mod.rs:83]: HTx looks like it has a "knee", which is a new field for the compressor

File: devtools/src/codegen/flexhtx/mod.rs (hunk 2 above)
Comment (mrene) [reply devtools/src/codegen/flexhtx/mod.rs:72]: Their XML files in the device console app has some comments around it, I think it's safe to ignore.
>

File: devtools/src/codegen/flexhtx/mod.rs (hunk 3 above)
Comment (mrene) [reply devtools/src/codegen/flexhtx/mod.rs:83]: It's there for other devices too (in the xml), but I've never seen it in the UI, so I assumed it was unreferenced.

File: devtools/src/codegen/flexhtx/mod.rs (hunk 1 above)
Comment (mrene) [reply devtools/src/codegen/flexhtx/mod.rs:51]: I assume these did not work? It looks like it's the right format.

**Comment by mrene** [comment]
Thanks for the PR and sorry for the radio silence

**Comment by danielhstahl** [comment]
NP, my bigger concern is the inability to actually create a binary for the RBPI...I'm hopeful that https://github.com/orgs/community/discussions/19197 will fix this by year end.

**Comment by mrene** [comment]
@danielhstahl try this out and let me know https://github.com/mrene/minidsp-rs/releases/tag/v0.1.10

**Comment by danielhstahl** [comment]
> @danielhstahl try this out and let me know https://github.com/mrene/minidsp-rs/releases/tag/v0.1.10

Beautiful!  Works perfectly, thank you!

**Pull request merged**